### PR TITLE
pacific: mgr/prometheus: use vendored "packaging" instead

### PR DESCRIPTION
**Pull request by Matan-B**
Backport of: #49712

Build: https://shaman.ceph.com/builds/ceph/wip-matanb-pacific-packaging-setuptools2/

Test Run: https://pulpito.ceph.com/matan-2023-01-11_14:16:28-smoke:basic-wip-matanb-pacific-packaging-setuptools2-distro-default-smithi/

Fixes: https://tracker.ceph.com/issues/58415

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Matan-B** [comment]
jenkins test api

**Review by ljflores — APPROVED** [review]

**Comment by ljflores** [comment]
@tchaikov we get this build error when trying to build this pacific backport:

```
Error: 
 Problem 1: package ceph-mgr-2:16.2.10-1623.ge4f26e5c.el8.x86_64 requires ceph-mgr-modules-core = 2:16.2.10-1623.ge4f26e5c.el8, but none of the providers can be installed
  - conflicting requests
  - nothing provides python3-packaging needed by ceph-mgr-modules-core-2:16.2.10-1623.ge4f26e5c.el8.noarch
```
https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/67215/consoleFull#145745529980e70a7f-704a-47c4-8937-06db83631476

Any suggestions on how we can tweak this to work with Pacific?

**Comment by tchaikov** [comment]
> @tchaikov we get this build error when trying to build this pacific backport:
> 
> ```
> Error: 
>  Problem 1: package ceph-mgr-2:16.2.10-1623.ge4f26e5c.el8.x86_64 requires ceph-mgr-modules-core = 2:16.2.10-1623.ge4f26e5c.el8, but none of the providers can be installed
>   - conflicting requests
>   - nothing provides python3-packaging needed by ceph-mgr-modules-core-2:16.2.10-1623.ge4f26e5c.el8.noarch
> ```
> 
> https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/67215/consoleFull#145745529980e70a7f-704a-47c4-8937-06db83631476
> 
> Any suggestions on how we can tweak this to work with Pacific?

@ljflores hi Laura, i believe we've done some research trying to address this issue. but other than the error message, i cannot find any details on the context, like what we've tried. since python3-packaging is provided[0] by the "powertools" repo, could you please check if this repo is enabled if we haven't done so?

---

[0] https://centos.pkgs.org/8/centos-powertools-x86_64/python3-packaging-16.8-9.el8.noarch.rpm.html

**Comment by tchaikov** [comment]
#49712 might help.

**Review by ljflores — DISMISSED** [review]

**Review by ljflores — APPROVED** [review]

**Comment by ljflores** [comment]
Hey @tchaikov, wondering if you have any insights on this.

We ran `upgrade-clients:client-upgrade-octopus-pacific` on the latest pacific RC, and it was failing consistently due to a packaging problem (i.e. https://pulpito.ceph.com/teuthology-2023-01-17_22:29:44-upgrade-clients:client-upgrade-octopus-pacific-pacific_16.2.11_RC6.6-distro-default-smithi/).

A few days ago, the same test that was scheduled automatically on a cron job was passing. I looked at the diff between the last working pacific branch and pacific RC, and there was only one commit difference: the one in this PR.
```
$ git log --pretty=oneline --no-merges 28d18736938371a6e23f3aba6f50cf6c1fc3fe4d..bcbf88bee4969f40f7fc319ee08e4d88e17faf44 src
0e4ead7bb8fa21f9e79ae56c07c992efee3492cb mgr/prometheus: use vendored "packaging" instead
```

As the test was failing from a ceph-mgr package, I thought this PR could be responsible (as the --suite-branch must be specified to octopus, so I thought maybe there was a package discrepancy between octopus and pacific).

I tried rerunning the test with this commit backported to octopus, and I got it to pass:
`./teuthology/virtualenv/bin/teuthology-suite -m smithi -c pacific_16.2.11_RC6.6 -s upgrade-clients:client-upgrade-octopus-pacific --suite-repo https://github.com/ljflores/ceph.git --suite-branch wip-octopus-backport -p 50`
http://pulpito.front.sepia.ceph.com/lflores-2023-01-19_23:54:51-upgrade-clients:client-upgrade-octopus-pacific-pacific_16.2.11_RC6.6-distro-default-smithi/

However, running the test normally, it is now passing:
`./teuthology/virtualenv/bin/teuthology-suite -m smithi -c pacific_16.2.11_RC6.6 -s upgrade-clients:client-upgrade-octopus-pacific --suite-repo https://github.com/ceph/ceph.git --suite-branch octopus -p 50`
http://pulpito.front.sepia.ceph.com/lflores-2023-01-19_23:54:51-upgrade-clients:client-upgrade-octopus-pacific-pacific_16.2.11_RC6.6-distro-default-smithi/

Long story short, I'm wondering if you can tell whether this commit was somehow responsible for the original failure (as it was failing consistently), and what might have changed.

**Comment by idryomov** [comment]
This is a backport of https://github.com/ceph/ceph/pull/49712 (trying to get Github to link this to the original PR properly).
